### ISSUE TITLE
Bug: MEM NoGui cannot be used if unable to connect to github even if it exists locally

### ISSUE THREAD

**Issue by Ravenleck** [ME3Tweaks/MassEffectModManager#395]
This is only an "issue" for me while the current v527 of Mass Effect Modder (MEM) is broken.

- problem: MEM v527 is broken.
- solution: downgrade to MEM v526*
- difficulty: the Manager either forcibly upgrades MEM no-GUI back to broken v527 or throws "host not found" error while trying to contact GitHub when you are offline

Either way has currently blocked me from installing any texture mods at all until MEM is fixed with a new patch. Yes, yes, I could still install texture mods manually using manual method with 526 MEM, but I've come to depend on texture installation through batch installer by now. :-) :-) :-) I'd have to rebuild my texture installation order in the file system, copying the order from my batch. Depending on how long it will take for MEM fix, I might still do that.

*for the GUI version of MEM, the Manager will skip update check when offline, as "expected". The "problem" is merely that the no-GUI version absolutely demands Internet access.

Log excerpt when I have MEM no-GUI v526 and am offline when starting a texture batch install follows. (It says "unknown host" in German, trying to resolve api.github.com.)

```
2023-11-27 14:15:05.234 +01:00 [INF] [M3] Submitted a background task to engine: TextureInstall
2023-11-27 14:15:05.266 +01:00 [INF] [ME3TWEAKSCORE] MassEffectModderNoGui launched, process ID: 13996
2023-11-27 14:15:05.317 +01:00 [INF] [ME3TWEAKSCORE] MassEffectModderNoGui exited normally with code 0
2023-11-27 14:15:05.317 +01:00 [INF] [ME3TWEAKSCORE] Checking for updates to MassEffectModderNoGui. The local version is 526
2023-11-27 14:15:05.319 +01:00 [ERR] [ME3TWEAKSCORE] An error occurred running MassEffectModderNoGui updater: 
2023-11-27 14:15:05.319 +01:00 [ERR] ME3TWEAKSCOREAggregateException: One or more errors occurred. (Der angegebene Host ist unbekannt. (api.github.com:443))
2023-11-27 14:15:05.319 +01:00 [ERR] ME3TWEAKSCORE   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)

2023-11-27 14:15:05.319 +01:00 [ERR] ME3TWEAKSCORE   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)

2023-11-27 14:15:05.319 +01:00 [ERR] ME3TWEAKSCORE   at System.Threading.Tasks.Task`1.get_Result()

2023-11-27 14:15:05.319 +01:00 [ERR] ME3TWEAKSCORE   at ME3TweaksCore.Helpers.MEM.MEMNoGuiUpdater.UpdateMEM(Boolean classicMEM, Boolean bypassSoakGate, Action`2 downloadProgressChanged, Action`1 exceptionUpdating, Action`1 statusMessageUpdate) in B:\UserProfile\source\repos\ME3Tweaks\MassEffectModManager\submodules\ME3TweaksCore\ME3TweaksCore\Helpers\MEM\MEMNoGuiUpdater.cs:line 60
2023-11-27 14:15:05.319 +01:00 [ERR] ME3TWEAKSCOREHttpRequestException: Der angegebene Host ist unbekannt. (api.github.com:443)
2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at System.Net.Http.HttpConnectionPool.AddHttp11ConnectionAsync(HttpRequestMessage request)

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at System.Net.Http.HttpConnectionPool.GetHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at System.Net.Http.DecompressionHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at Octokit.Internal.HttpClientAdapter.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in /_/Octokit/Http/HttpClientAdapter.cs:line 199

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at Octokit.Internal.HttpClientAdapter.Send(IRequest request, CancellationToken cancellationToken, Func`2 preprocessResponseBody) in /_/Octokit/Http/HttpClientAdapter.cs:line 50

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at Octokit.Connection.RunRequest(IRequest request, CancellationToken cancellationToken, Func`2 preprocessResponseBody) in /_/Octokit/Http/Connection.cs:line 704

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at Octokit.Connection.Run[T](IRequest request, CancellationToken cancellationToken, Func`2 preprocessResponseBody) in /_/Octokit/Http/Connection.cs:line 695

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at Octokit.ApiConnection.GetPage[TU](Uri uri, IDictionary`2 parameters, String accepts, ApiOptions options, Func`2 preprocessResponseBody) in /_/Octokit/Http/ApiConnection.cs:line 672

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at Octokit.ApiConnection.<>c__DisplayClass19_0`1.<<GetAll>b__0>d.MoveNext() in /_/Octokit/Http/ApiConnection.cs:line 230

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE--- End of stack trace from previous location ---

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at Octokit.ApiPagination.GetAllPages[T](Func`1 getFirstPage, Uri uri)
2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORESocketException: Der angegebene Host ist unbekannt.
2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at System.Net.Sockets.Socket.<ConnectAsync>g__WaitForConnectWithCancellation|277_0(AwaitableSocketAsyncEventArgs saea, ValueTask connectTask, CancellationToken cancellationToken)

2023-11-27 14:15:05.320 +01:00 [ERR] ME3TWEAKSCORE   at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)
2023-11-27 14:15:05.320 +01:00 [ERR] [M3] Failed to extract MassEffectModderNoGui:
2023-11-27 14:15:05.320 +01:00 [ERR] [M3] AggregateException: One or more errors occurred. (Der angegebene Host ist unbekannt. (api.github.com:443))
2023-11-27 14:15:05.320 +01:00 [ERR] [M3]    at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)

2023-11-27 14:15:05.320 +01:00 [ERR] [M3]    at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)

2023-11-27 14:15:05.320 +01:00 [ERR] [M3]    at System.Threading.Tasks.Task`1.get_Result()

2023-11-27 14:15:05.320 +01:00 [ERR] [M3]    at ME3TweaksCore.Helpers.MEM.MEMNoGuiUpdater.UpdateMEM(Boolean classicMEM, Boolean bypassSoakGate, Action`2 downloadProgressChanged, Action`1 exceptionUpdating, Action`1 statusMessageUpdate) in B:\UserProfile\source\repos\ME3Tweaks\MassEffectModManager\submodules\ME3TweaksCore\ME3TweaksCore\Helpers\MEM\MEMNoGuiUpdater.cs:line 60
2023-11-27 14:15:05.320 +01:00 [ERR] [M3] HttpRequestException: Der angegebene Host ist unbekannt. (api.github.com:443)
2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at System.Net.Http.HttpConnectionPool.AddHttp11ConnectionAsync(HttpRequestMessage request)

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at System.Net.Http.HttpConnectionPool.GetHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at System.Net.Http.DecompressionHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at Octokit.Internal.HttpClientAdapter.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in /_/Octokit/Http/HttpClientAdapter.cs:line 199

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at Octokit.Internal.HttpClientAdapter.Send(IRequest request, CancellationToken cancellationToken, Func`2 preprocessResponseBody) in /_/Octokit/Http/HttpClientAdapter.cs:line 50

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at Octokit.Connection.RunRequest(IRequest request, CancellationToken cancellationToken, Func`2 preprocessResponseBody) in /_/Octokit/Http/Connection.cs:line 704

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at Octokit.Connection.Run[T](IRequest request, CancellationToken cancellationToken, Func`2 preprocessResponseBody) in /_/Octokit/Http/Connection.cs:line 695

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at Octokit.ApiConnection.GetPage[TU](Uri uri, IDictionary`2 parameters, String accepts, ApiOptions options, Func`2 preprocessResponseBody) in /_/Octokit/Http/ApiConnection.cs:line 672

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at Octokit.ApiConnection.<>c__DisplayClass19_0`1.<<GetAll>b__0>d.MoveNext() in /_/Octokit/Http/ApiConnection.cs:line 230

2023-11-27 14:15:05.321 +01:00 [ERR] [M3] --- End of stack trace from previous location ---

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at Octokit.ApiPagination.GetAllPages[T](Func`1 getFirstPage, Uri uri)
2023-11-27 14:15:05.321 +01:00 [ERR] [M3] SocketException: Der angegebene Host ist unbekannt.
2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource.GetResult(Int16 token)

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at System.Net.Sockets.Socket.<ConnectAsync>g__WaitForConnectWithCancellation|277_0(AwaitableSocketAsyncEventArgs saea, ValueTask connectTask, CancellationToken cancellationToken)

2023-11-27 14:15:05.321 +01:00 [ERR] [M3]    at System.Net.Http.HttpConnectionPool.ConnectToTcpHostAsync(String host, Int32 port, HttpRequestMessage initialRequest, Boolean async, CancellationToken cancellationToken)
2023-11-27 14:15:05.321 +01:00 [ERR] [ME3TWEAKSCORE] Exception occurred in TextureInstaller thread: Could not download MassEffectModderNoGui - texture installation cannot proceed. See Mod Manager logs for more information.
2023-11-27 14:15:05.321 +01:00 [ERR] [ME3TWEAKSCORE]    at ME3TweaksModManager.modmanager.usercontrols.TextureInstallerPanel.<OnPanelVisible>b__17_0(Object a, DoWorkEventArgs b) in B:\UserProfile\source\repos\ME3Tweaks\MassEffectModManager\MassEffectModManagerCore\modmanager\usercontrols\TextureInstallerPanel.xaml.cs:line 221
   at System.ComponentModel.BackgroundWorker.OnDoWork(DoWorkEventArgs e)
   at ME3TweaksCore.Helpers.NamedBackgroundWorker.OnDoWork(DoWorkEventArgs e) in B:\UserProfile\source\repos\ME3Tweaks\MassEffectModManager\submodules\ME3TweaksCore\ME3TweaksCore\Helpers\NamedBackgroundWorker.cs:line 38
   at System.ComponentModel.BackgroundWorker.WorkerThreadStart(Object argument)
2023-11-27 14:15:05.321 +01:00 [INF] [M3] Completed a background task: TextureInstall
```

**Comment by Mgamerz** [comment]
This has been committed to the codebase and will be available in Mod Manager 9.

**Comment by Mgamerz** [comment]
Implemented in 9.0
